src/Chapter1.hs

```diff
@@ -542,7 +543,11 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z  
+  | (x >= y && x <= z) || (x >= z && x <= z) = x
+  | (y >= x && y <= z) || (y >= z && y <= z) || (y >= z && y <= x) = y
```

Comment:
I wonder, is this second check necessary :thinking: You probably thought about the implementation more than me, but at first glance, it seems redundant.

```suggestion
  | (y >= x && y <= z) || (y >= z && y <= x) = y
```